src/black/concurrency.py

```diff
@@ -171,7 +172,11 @@ async def schedule_formatting(
             if task.cancelled():
                 cancelled.append(task)
             elif task.exception():
-                report.failed(src, str(task.exception()))
+                exc = task.exception()
+                assert exc is not None
```

Comment:
Use a walrus so you can avoid the assert?

`elif exc := task.exception():`

Reply:
Oh right, we dropped 3.7